sdk/azidentity/device_code_credential.go

```diff
@@ -15,27 +16,47 @@ const (
 	deviceCodeGrantType = "urn:ietf:params:oauth:grant-type:device_code"
 )
 
+type DeviceCodeCredentialOptions struct {
+	TenantID   *string                 // Gets the Azure Active Directory tenant (directory) ID of the service principal
```

Comment:
I believe `TenantID` and `ClientID` won't be pointer-to-string.

Reply:
Done